compiler/src/dotty/tools/dotc/core/Symbols.scala

```diff
@@ -516,6 +516,18 @@ object Symbols {
     /** The current name of this symbol */
     final def name(implicit ctx: Context): ThisName = denot.name.asInstanceOf[ThisName]
 
+    def isHigherOrderTypeParameter(implicit ctx: Context): Boolean = this.maybeOwner.isTypeParameterOrSkolem
+    def isTypeParameterOrSkolem(implicit ctx: Context): Boolean = this.isTypeParam
+    def enclClassChain(implicit ctx: Context): List[Symbol] = this.maybeOwner.enclClassChain
+    final def isDerivedValueClass(implicit ctx: Context): Boolean =
+      isClass && !denot.is(Flags.Package) && !denot.is(Flags.Trait) &&
```

Comment:
No need to write `denot`. It's used as an implicit everywhere else.

Reply:
Symbols.scala, as most source files, imports all flags with a wildcard. So for consistency I would not use the `Flags` prefix.

Reply:
You can combine the two tests like this:

    !this.is(Package | Trait)

Reply:
But in the end, you should just use `ValueClasses.isDerivedValueClass`. If the two definitions differ, we should find out which is the right one and use only that one.